### PR TITLE
Istio with slirp Interface POC

### PR DESCRIPTION
**Pull request by SchSeba**
This is a POC of using istio with kubevirt.

Don't merge until [issue-1211](https://github.com/kubevirt/kubevirt/issues/1211) is closed.

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/kubevirt.github.io/62)

### CODE REVIEW

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 1]
```diff
@@ -0,0 +1,253 @@
+---
+layout: post
+author: SchSeba
+description: Use istio with kubevirt
+---
+
+# Introduction
+
+On this blog post, we are going to deploy virtual machines with the kubevirt project and insert them into the istio service mesh.
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: KubeVirt

Istio

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+This demo is going to be deployed on a k8s 1.10 cluster.
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: Kubernetes

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+
+# Requirements
+
+* docker
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: docker - For …

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+* kubeadm
+
+Follow this [document](https://kubernetes.io/docs/tasks/tools/install-kubeadm/) to install everything we need for the POC
+
+
+# Deployment
+
+For the POC we clone [this repo](https://github.com/SchSeba/kubevirt-istio-poc)
+
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: Please explain in a paragraph what this repo does.

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+Run the bash script
+
+```
+cd kubevirt-istio-poc
+./deploy-istio-poc.sh
+```
+
+# Demo application
+
+We are going to use the [bookinfo sample application](https://istio.io/docs/guides/bookinfo/#overview) from the istio webpage
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: We can end the sentence with a dot (.)

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+The follow yaml will deploy the bookinfo application with a 'small' change the details service will run on a virtual machine!
+
+<span style="color:blue;">Note: it will take like 5 minutes for the application to by running inside the virtual machine because we install git and ruby, then clone the istio repo and start the application</span>
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: You can use:

`> **Note:** …`

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+Lets start with the bash script:
+
+```
+#!/bin/bash
+
+set -x
+
+kubeadm init --pod-network-cidr=192.168.0.0/16
+
+yes | cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: You probably require root access for this?

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+kubectl apply -f istio-demo-auth.yaml
+
+echo Add istio-injection to the default namespace.
+
+kubectl label namespace default istio-injection=enabled 
+
+
+ready=$(kubectl get po -n istio-system | grep sidecar-injector | grep Running | wc -l)
+
+while [ $ready -eq 0 ]
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: you can do

```
while [[ $(kubectl get po -n istio-system | grep sidecar-injector | grep Running | wc -l) -eq ]];
…
```

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+  networks:
+  - name: testSlirp
+    pod: {}
+  terminationGracePeriodSeconds: 0
+  volumes:
+  - name: registryvolume
+    registryDisk:
+      image: kubevirt/fedora-cloud-registry-disk-demo:latest
+  - cloudInitNoCloud:
+      userData: |-
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: REALLY neat idea!

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+      disks:
+      - disk:
+          bus: virtio
+        name: registrydisk
+        volumeName: registryvolume
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+        volumeName: cloudinitvolume
+      interfaces:
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: Will one of your scripts above build kubevirt from the branch which has your patch?

or should we wait for your PR to get merged?

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown (hunk 2 above)
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: Is this deploying Kubernetes or should Kubernetes be already set up?

What KubeVirt version will be used?

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 11]
```diff
@@ -0,0 +1,253 @@
+---
+layout: post
+author: SchSeba
+description: Use istio with kubevirt
+---
+
+# Introduction
+
+On this blog post, we are going to deploy virtual machines with the kubevirt project and insert them into the istio service mesh.
+
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: Please give a broader overview over the components k8s, kubevitr, calico, istio, book application. Just like a few lines.

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+reviews-v1-64545d97b4-tvgl2       2/2       Running   0          1h
+reviews-v2-8cb9489c6-wjp9x        2/2       Running   0          1h
+reviews-v3-6bc884b456-hr5bm       2/2       Running   0          1h
+virt-launcher-vmi-details-94pb6   3/3       Running   0          1h
+```
+
+Lets find the istio ingress service port
+
+```
+# kubectl get service -n istio-system  | grep istio-ingressgateway
```
Comment (fabiand) [inline _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: Is there an istioctl command that could be used?

Comment (SchSeba) [reply _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: No there is no Istio command that I know about it to check if the deployment is finished.

File: _posts/2019-06-21-Run-Istio-with-kubevirt.markdown (hunk 4 above)
Comment (SchSeba) [reply _posts/2019-06-21-Run-Istio-with-kubevirt.markdown]: Done

**Comment by scollier** [comment]
LGTM. Sorry for the delay. Excellent post.

**Pull request merged**